Fix and re-enable some IPC LayoutTests

jon-b-apple opened the pull request:
#### 7d7750b55a1bd06b8e7b88d9261ce58e03e81ccf
<pre>
Fix and re-enable some IPC LayoutTests
<a href="https://rdar.apple.com/162864398">rdar://162864398</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=300983">https://bugs.webkit.org/show_bug.cgi?id=300983</a>

Reviewed by Alex Christensen.

Updated some IPC LayoutTests to fix errors and crashes that were preventing them from running.

Aside from updating the test contents, this required updating the IPC Testing API wrapper around stream connections to invalidate
the connection on finalize (if needed), avoiding a debug assertion in the StreamClientConnection destructor.

* LayoutTests/TestExpectations:
* LayoutTests/ipc/invalid-feConvolveMatrix-crash.html:
* LayoutTests/ipc/media-player-invalid-test.html:
* LayoutTests/ipc/restore-empty-stack-crash.html:
* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::finalize):

Canonical link: <a href="https://commits.webkit.org/302384@main">https://commits.webkit.org/302384@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e2a97cc97fb007ebb5a466ae94f3b97e28855936

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127380 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/47028 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38164 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134444 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78935 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6b9ab70e-add0-424a-9b37-4708e6abcef0) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47643 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55551 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96944 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64934 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/821b7014-0027-4ab9-8730-af12b85f5b83) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130328 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/38128 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/114090 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77438 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a5fb1c1a-a4ba-4bf5-aabd-e31028034299) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36971 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/32203 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77826 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107987 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32611 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136927 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/54039 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41644 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105466 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54550 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110439 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/105142 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27117 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50662 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30402 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/51618 "The change is no longer eligible for processing.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53976 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/60063 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/53210 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56667 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54969 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/Cocoa/CoreIPCPassKit.serialization.in:
```diff
@@ -86,7 +86,9 @@ webkit_platform_headers: "CoreIPCPKSecureElementPass.h"
     useLargeFont: Number
     identifier: String
     detail: String
+#if !HAVE(WK_SECURE_CODING_PKDATECOMPONENTSRANGE)
```
This is a change to functional behavior, and I think it's incorrect.  If HAVE(WK_SECURE_CODING_PKDATECOMPONENTSRANGE) is true, we serialize it using CoreIPCPKDateComponentsRange.  If it's false, we serialize it using webkit_secure_coding.  We shouldn't just remove the member on new OSes.  That would certainly break something.

jon-b-apple replied:
Thanks for spotting this, this was a lot more complex than it appeared. I had several goes at getting it working for both HAVE and !HAVE cases, and while I did eventually succeed, it was by deleting some code that was added a week ago in https://github.com/WebKit/WebKit/pull/51778.

For now, I removed the change here, and I will follow up with Simon to see why we implemented it the way we did, and whether the test passed after the change was made.

achristensen07 commented on LayoutTests/TestExpectations:
```diff
@@ -5986,12 +5986,11 @@ imported/w3c/web-platform-tests/css/motion/offset-path-url-009.html [ ImageOnlyF
+[ Debug ] ipc/send-invalid-message.html [ Pass ]
 
-webkit.org/b/297616 [ Debug ] ipc/invalid-feConvolveMatrix-crash.html [ Crash ]
+[ Debug ] ipc/invalid-feConvolveMatrix-crash.html [ Pass ]
```
We should probably remove the test expectations for these files entirely if they pass now.

achristensen07 commented on LayoutTests/platform/mac-wk2/TestExpectations:
```diff
@@ -2043,7 +2043,7 @@ webkit.org/b/288660 [ Sonoma Release x86_64 ] media/media-hevc-video-as-img.html
 [ Sonoma Release x86_64 ] imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-below-viewport-offscreen-old.html [ Skip ]
 
-webkit.org/b/288742 [ Debug ] ipc/media-player-invalid-test.html [ Crash ]
+[ Debug ] ipc/media-player-invalid-test.html [ Pass ]
```
ditto.

jon-b-apple committed:
Fix and re-enable some IPC LayoutTests
rdar://162864398
https://bugs.webkit.org/show_bug.cgi?id=300983

Reviewed by Alex Christensen.

Updated some IPC LayoutTests to fix errors and crashes that were preventing them from running.

Aside from updating the test contents, this required updating the IPC Testing API wrapper around stream connections to invalidate
the connection on finalize (if needed), avoiding a debug assertion in the StreamClientConnection destructor.

* LayoutTests/TestExpectations:
* LayoutTests/ipc/invalid-feConvolveMatrix-crash.html:
* LayoutTests/ipc/media-player-invalid-test.html:
* LayoutTests/ipc/restore-empty-stack-crash.html:
* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::finalize):

Canonical link: https://commits.webkit.org/302384@main
